### DISCUSSION TITLE
CMakeLists.txt for Pimoroni Plasma 2040 (GPIO14-15) with Multi-Segment

### DISCUSSION THREAD

**Discussion by robertvb83** [awawa-dev/HyperSerialPico#19] [Q&A] [answered]
Hi,

i would like to compile the firmware for the Pimoroni Plasma 2040 for use in a new (reversed) multi-segment setup with 600 LEDs and i have a few questions on changing the CMakeLists.txt

1. What is the index of the first LED? is it 0 or 1. In the description you mention the second segment starts at INDEX 300, i would have thought it must be 301... (First segment being 1-300)

> 600LEDs RGBW
> SECOND_SEGMENT_INDEX=300

2. The Pimoroni Plasma 2040 default Output pins are 14 (CL) and 15 (DA). In single-segment, 15 is used as the default data output. And the description says 

> If multi-segment mode is enabled for SK6812/WS281x, the output for the second segment is always the next GPIO pin (OUTPUT_DATA_PIN + 1)

does this mean, that for multi-segment the data pin needs to be set to 14 so that the second one is 14+1=15 and does this mean, the first LED section needs to be connected to 14 and the second segment to 15?

`set(OUTPUT_DATA_PIN 14)`

Long story short, would you change anything on my attempt to change the CMakeLists.txt for Pimoroni Plasma 2040?

```
if (NOT PICO_PROGRAM_MAIN_ENTRY)
	# User configuration for HyperSerialPico section starts here

	# Some boards, such as the first Adafruit revisions, may have trouble booting properly
	# due to bad componets used in the design.
	# Turn this setting to ON if your rp2040 is not detected after firmware upload and reset
	set(BOOT_WORKAROUND OFF)

	# Default output data pin for the non-SPI LED strips (only for sk6812/ws2812b)
	set(OUTPUT_DATA_PIN 14)

	# Default output data/clocks pin for the SPI LED strips (only for apa102, not for sk6812/ws2812b)
	# only certain pairs of pins are allowed for selected SPI interface (refer to "readme.md")
	set(OUTPUT_SPI_DATA_PIN 15)
	set(OUTPUT_SPI_CLOCK_PIN 14)
	set(OUTPUT_SPI_INTERFACE spi1)

	# Use multi-segment, starting index of second led strip or OFF to disable
	set(SECOND_SEGMENT_INDEX 301)

	# If multi-segment is used and it's reversed, set this option to ON to enable reversing
	set(SECOND_SEGMENT_REVERSED ON)

	# User configuration section ends here
	# Usually you don't need to change anything below this section
endif()
```

**Comment by awawa-dev** [comment, marked as answer]
Hi
> What is the index of the first LED? is it 0 or 1. In the description you mention the second segment starts at INDEX 300, i would have thought it must be 301... (First segment being 1-300)

First LED has 0 index. so first 300 segment is 0-299 and the second segment is 300-599. Total 600LEDs configured in HyperHDR.

>> If multi-segment mode is enabled for SK6812/WS281x, the output for the second segment is always the next GPIO pin (OUTPUT_DATA_PIN + 1)
>
> does this mean, that for multi-segment the data pin needs to be set to 14 so that the second one is 14+1=15 and does this mean, the first LED section needs to be connected to 14 and the second segment to 15?

Yes, this rule about the second segment (pin +1) always applies, so the first segment is OUTPUT_DATA_PIN  = 14 (the second will be 15)

> Long story short, would you change anything on my attempt to change the CMakeLists.txt for Pimoroni Plasma 2040?

set(SECOND_SEGMENT_INDEX 301) -> set(SECOND_SEGMENT_INDEX 300) ?

You didnt need to change SPI settings, it is simply ignored for Neopixel LEDs: OUTPUT_SPI_INTERFACE, OUTPUT_SPI_DATA_PIN, OUTPUT_SPI_CLOCK_PIN.

**Reply by robertvb83** [reply]
Thank you! That perfectly clarifies my questions! Now i am on fire getting this setup tested.

My plan is to switch from my APA102 hyperion NG configuration to hyperHDR

**Reply by robertvb83** [reply]
is it safe to ignore that warning message after compiling?

> **Annotations**
> 1 warning
> [HyperSerialPico](https://github.com/robertvb83/HyperSerialPico/actions/runs/7786031503/job/21229911964)
> Node.js 16 actions are deprecated. Please update the following actions to use Node.js 20: actions/checkout@v3, actions/upload-artifact@v3. For more information see: https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/.

**Reply by awawa-dev** [reply]
Yes, ignore it. it's nothing important
